micro-deps/micro-deps-spring-config/src/test/groovy/com/ofg/infrastructure/discovery/ApplicationContextStartupSpec.groovy

```diff
@@ -2,18 +2,27 @@ package com.ofg.infrastructure.discovery
 
 import com.ofg.infrastructure.discovery.config.PropertySourceConfiguration
 import com.ofg.infrastructure.discovery.watcher.presence.checker.NoInstancesRunningException
+import com.ofg.stub.server.AvailablePortScanner
 import org.apache.curator.test.TestingServer
 import org.springframework.context.annotation.AnnotationConfigApplicationContext
 import spock.lang.Specification
+import spock.util.environment.RestoreSystemProperties
 
 import static com.ofg.config.BasicProfiles.PRODUCTION
 import static org.codehaus.groovy.runtime.StackTraceUtils.extractRootCause
 
 class ApplicationContextStartupSpec extends Specification {
 
+    @RestoreSystemProperties
     def 'should fail to start application context if resource is missing when default bean is missing deps'() {
         given:
-            TestingServer testingServer = new TestingServer(2181)
+            TestingServer testingServer
+            Integer port
+            new AvailablePortScanner(9000, 9500).tryToExecuteWithFreePort { int freePort ->
+                testingServer = new TestingServer(freePort)
+                port = freePort
+            }
+            System.setProperty('service.resolver.url', "localhost:$port")
```

Comment:
Done in https://github.com/4finance/micro-infra-spring/commit/41841e01cdf533387f702c81aafdfa74535a2a6e